### ISSUE TITLE
posterior_predict doesn't work for stan_polr when outcome has 2 levels

### ISSUE THREAD

**Issue by jgabry** [stan-dev/rstanarm#81]
@bgoodri Had this been working before? There's no test for it and I can't imagine it was ever working because the line

```{r}
zeta <- stanmat[, grep("|", colnames(stanmat), value = TRUE, fixed = TRUE)]
```

is not applicable if there are no cutpoints but is called regardless of whether or not there are cutpoints.

**Comment by bgoodri** [comment]
Yeah, it should call posterior_predict for the Bernoulli case.

**Comment by jgabry** [comment]
What about the scobit models?

**Comment by jgabry** [comment]
So the issue for the Bernoulli case is that stan_polr was returning an object with class "polr" regardless and so the posterior_predict method for stan_polr models was being used instead of posterior_predict for binomial models.

**Comment by jgabry** [comment]
Can we entirely remove "polr" from the classes of all stan_polr models with 2-level outcomes? Or do we need it to have class "polr" for some other reasons?

**Comment by jgabry** [comment]
I think you explicitly adding some things to .pp_polr for the scobit models so I don't think it can use the method for binomial models

**Comment by jgabry** [comment]
I fixed the problem for the non-scobit models in 047c8e5. So I think it's just the scobit case that still needs to be fixed.

**Comment by bgoodri** [comment]
OK. I will fix the scobit case.

**Comment by bgoodri** [comment]
Scobit case fixed by e19b4cceacb02e136aab83a292bc2f3b0fba8456
